akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -635,9 +611,18 @@ private[stream] abstract class MaterializerSession(val topLevel: StreamLayout.Mo
       s"The top level module cannot be materialized because it has unconnected ports: ${(topLevel.inPorts ++ topLevel.outPorts).mkString(", ")}")
     try materializeModule(topLevel, topLevel.attributes)
     catch {
-      case NonFatal(e) ⇒
-        panic(e)
-        throw e
+      case NonFatal(cause) ⇒
+        // Cancels all intermediate Publishers and fails all intermediate Subscribers.
+        // (This is an attempt to clean up after an exception during materialization)
+        val panicError = new MaterializationPanic(cause)
```

Comment:
@drewhk I embedded "panic" here since it wasn't used anywhere else. But to me, it's hard to reason about if and why it is correct.